### ISSUE TITLE
'emv search' finds same AID multiple times 

### ISSUE THREAD

**Issue by pwpiwi** [Proxmark/proxmark3#767]
```
proxmark3> emv search -s
 Search completed.
|------------------|--------|-------------------------|
|    AID           |Priority| Name                    |
|------------------|--------|-------------------------|
|a000000025010801  |   01   |AMERICAN EXPRESS         |
|a000000025010801  |   01   |AMERICAN EXPRESS         |
|a000000025010801  |   01   |AMERICAN EXPRESS         |
|------------------|--------|-------------------------|
proxmark3>
```
Probably caused by returning AID which only partially match. Searching for "A000000025", "A00000002501" and "A000000025010801" will all find the latter AID.

**Comment by merlokk** [comment]
it looks that PSE have several records
may you put here result `emv pse -st` ?

**Comment by pwpiwi** [comment]
'emv search' doesn't check the PPSE but tries each AID from the list. See the result of 'emv search -st':
```
proxmark3> emv search -sat
 >>>> 00 a4 04 00 09 a0 00 00 00 03 05 07 60 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 03 10 10 01 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 03 10 10 02 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 20 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 20 20 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 30 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 40 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 50 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 60 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 60 20 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 80 02 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 80 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 03 90 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 03 99 99 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 05 a0 00 00 00 98 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 98 08 48 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 06 a0 00 00 00 04 01 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 09 a0 00 00 00 04 10 10 12 13 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 09 a0 00 00 00 04 10 10 12 15 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 20 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 30 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 30 60 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 04 30 60 01 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 40 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 50 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 60 00 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 80 02 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 04 99 99 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 05 a0 00 00 00 25 00
 <<<< 6f 2c 84 08 a0 00 00 00 25 01 08 01 a5 20 50 10 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 9f 38 03 9f 35 01 87 01 01 5f 2d 02 64 65 90 00
 A000000025
-------------------- TLV decoded --------------------
--6f[2c] 'File Control Information (FCI) Template':
        --84[08] 'Dedicated File (DF) Name':
                00: a0 00 00 00 25 01 08 01                         |....%...
        --a5[20] 'File Control Information (FCI) Proprietary Template':
                --50[10] 'Application Label':   String value 'AMERICAN EXPRESS'
                        00: 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 |AMERICAN EXPRESS
                --9f38[03] 'Processing Options Data Object List (PDOL)':
                        Tag 9f35 len 01 ('Terminal Type')
                        00: 9f 35 01                                        |.5.
                --87[01] 'Application Priority Indicator':
                        00: 01                                              |.
                --5f2d[02] 'Language Preference':       String value 'de'
                        00: 64 65                                           |de
 >>>> 00 a4 04 00 07 a0 00 00 00 25 00 00 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 06 a0 00 00 00 25 01 00
 <<<< 6f 2c 84 08 a0 00 00 00 25 01 08 01 a5 20 50 10 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 9f 38 03 9f 35 01 87 01 01 5f 2d 02 64 65 90 00
 A00000002501
-------------------- TLV decoded --------------------
--6f[2c] 'File Control Information (FCI) Template':
        --84[08] 'Dedicated File (DF) Name':
                00: a0 00 00 00 25 01 08 01                         |....%...
        --a5[20] 'File Control Information (FCI) Proprietary Template':
                --50[10] 'Application Label':   String value 'AMERICAN EXPRESS'
                        00: 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 |AMERICAN EXPRESS
                --9f38[03] 'Processing Options Data Object List (PDOL)':
                        Tag 9f35 len 01 ('Terminal Type')
                        00: 9f 35 01                                        |.5.
                --87[01] 'Application Priority Indicator':
                        00: 01                                              |.
                --5f2d[02] 'Language Preference':       String value 'de'
                        00: 64 65                                           |de
 >>>> 00 a4 04 00 08 a0 00 00 00 25 01 04 02 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 25 01 07 01 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 08 a0 00 00 00 25 01 08 01 00
 <<<< 6f 2c 84 08 a0 00 00 00 25 01 08 01 a5 20 50 10 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 9f 38 03 9f 35 01 87 01 01 5f 2d 02 64 65 90 00
 A000000025010801
-------------------- TLV decoded --------------------
--6f[2c] 'File Control Information (FCI) Template':
        --84[08] 'Dedicated File (DF) Name':
                00: a0 00 00 00 25 01 08 01                         |....%...
        --a5[20] 'File Control Information (FCI) Proprietary Template':
                --50[10] 'Application Label':   String value 'AMERICAN EXPRESS'
                        00: 41 4d 45 52 49 43 41 4e 20 45 58 50 52 45 53 53 |AMERICAN EXPRESS
                --9f38[03] 'Processing Options Data Object List (PDOL)':
                        Tag 9f35 len 01 ('Terminal Type')
                        00: 9f 35 01                                        |.5.
                --87[01] 'Application Priority Indicator':
                        00: 01                                              |.
                --5f2d[02] 'Language Preference':       String value 'de'
                        00: 64 65                                           |de
 >>>> 00 a4 04 00 07 a0 00 00 00 42 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 42 20 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 42 30 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 42 40 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 42 50 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 06 a0 00 00 00 65 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 00 65 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 01 54 44 42 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 f0 00 00 00 03 00 01 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 a0 00 00 05 24 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 >>>> 00 a4 04 00 07 d5 78 00 00 02 10 10 00
 <<<< 6a 82
ERROR: APDU(00a4) ERROR: [6A82] File not found
 Search completed.
proxmark3>
```
BTW: doesn't print the table with '-sat' ???

**Comment by merlokk** [comment]
yes, you are right) needs to check if field exists.

it not prints table with `-t`

**Comment by pwpiwi** [comment]
fixed by PR #776
